File: dato.config.rb
directory "data/" do
  create_data_file "home.toml", :toml,
    site_name: dato.homepage.site_name,
    tag_line: dato.homepage.tag_line,
    description: dato.homepage.description
end

directory "content/season" do
  dato.seasons.each do |season|
    create_post "#{season.slug}.md" do
      frontmatter :yaml,
        title: season.name,
        image_url: season.image.file.width(400).to_url,
        thumbnail_url: season.image.file.height(300).to_url,
        bg_url: season.image.file.width(5).to_url,
        weight: season.position,
        excerpt: season.overview.truncate_words(50)

      content season.overview
    end
  end
end

directory "content/episode" do
  dato.episodes.sort_by { |e| [e.season.id.to_i, e.episode_number] }.each_with_index do |episode, i|
    create_post "#{episode.slug}.md" do
      frontmatter :toml,
        title: episode.title,
        episodenumber: episode.episode_number,
        palette_url: episode.image && episode.image.file.auto("enhance").palette("json").to_url,
        imageurl: episode.image && episode.image.file.to_url,
        thumbnail_url: episode.image && episode.image.file.width(500).height(280).fit("crop").auto("enhance").format(:jpg).to_url,
        date: episode.first_aired.to_s,
        weight: i,
        rating: episode.rating,
        director: episode.director,
        category: episode.season.name

      content episode.description
    end
  end
end

directory "content/character" do
  dato.characters.each do |character|
    create_post "#{character.slug}.md" do
      frontmatter :toml,
        title: character.name,
        actorname: character.actor_name,
        episodes: character.episode,
        weight: character.position,
        thumbnail_url: character.image.file.fit('crop').crop('faces').width(200).height(200).to_url,
        image_url: character.image.file.width(500).format(:jpg).to_url

      content character.description
    end
  end
end

add_to_data_file "config.toml", :toml,
  title: dato.homepage.site_name

File: dato.config.js
const ellipsize = require("ellipsize");

module.exports = (dato, root) => {
  root.directory("data/", (dir) => {
    dir.createDataFile("home.toml", "toml", {
      sitename: dato.homepage.siteName,
      tagline: dato.homepage.tagLine,
      description: dato.homepage.description,
    });
  });

  root.directory("content/season", (dir) => {
    dato.seasons.forEach((season) => {
      dir.createPost(`${season.slug}.md`, "yaml", {
        frontmatter: {
          title: season.name,
          imageurl: season.image.url({ w: 400 }),
          thumbnailurl: season.image.url({ h: 300 }),
          bgurl: season.image.url({ w: 5 }),
          weight: season.position,
          excerpt: ellipsize(season.overview, 150),
        },
        content: season.overview,
      });
    });
  });

  root.directory("content/episode", (dir) => {
    dato.episodes
      .sort((a, b) => {
        return b.id - a.id;
      })
      .forEach((episode, i) => {
        dir.createPost(`${episode.slug}.md`, "toml", {
          frontmatter: {
            title: episode.title,
            episodenumber: episode.episodeNumber,
            paletteurl:
              episode.image &&
              episode.image.url({ auto: "enhance", palette: "json" }),
            imageurl: episode.image && episode.image.url(),
            thumbnailurl:
              episode.image &&
              episode.image.url({
                w: 500,
                h: 280,
                fit: "crop",
                auto: "enhance",
                fm: "jpg",
              }),
            date: episode.firstAired.toMap(),
            weight: i,
            rating: episode.rating,
            director: episode.director,
            category: episode.season.name,
          },
          content: episode.description,
        });
      });
  });

  root.directory("content/character", (dir) => {
    dato.characters.forEach((character) => {
      dir.createPost(`${character.slug}.md`, "toml", {
        frontmatter: {
          title: character.name,
          actorname: character.actorName,
          episodes: character.episode,
          weight: character.position,
          thumbnailurl: character.image.url({
            fit: "crop",
            crop: "faces",
            w: 200,
            h: 200,
          }),
          imageurl: character.image.url({ w: 500, fm: "jpg" }),
        },
        content: character.description,
      });
    });
  });

  root.addToDataFile("config.toml", "toml", { title: dato.homepage.siteName });
};
